conceptnet5/vectors/transforms.py

```diff
@@ -101,3 +145,64 @@ def make_replacements(small_frame, big_frame, limit=100000):
             if got:
                 replacements[term] = got[0]
     return replacements
+
+
+def choose_small_vocabulary(big_frame, concepts_filename):
+    """
+    Choose the vocabulary of the small frame, by eliminating the terms which:
+     - contain more than one word
+     - are not in ConceptNet
+    """
+    concepts = set(line.strip() for line in open(concepts_filename))
+    small_vocab = [term for term in big_frame.index if term.count('_') < 1 and term in concepts]
+    return small_vocab
+
+
+def make_big_frame(frame, language):
+    """
+     Choose the vocabulary for the big frame and make the big frame. Eliminate the terms which
+     are in languages other than the language specified.
+    """
+    vocabulary = [term for term in frame.index if get_language(term) == language]
+    big_frame = frame.ix[vocabulary]
+    return big_frame
+
+
+def make_small_frame(big_frame, concepts_filename):
+    """
+    Create a small frame using the output of choose_small_vocabulary()
+    """
+    small_vocab = choose_small_vocabulary(big_frame, concepts_filename)
+    return big_frame.ix[small_vocab]
+
+
+def save_replacements(output_filepath, replacements):
+    # Save the replacement dictionary as a mgspack file
+    with open(output_filepath, 'wb') as output_file:
+        msgpack.dump(replacements, output_file)
+
+
+def save_frame(output_filepath, frame):
+    frame.to_hdf(output_filepath, 'mat', encoding='utf-8')
+
+
+@click.command()
+@click.argument('frame-filename')
+@click.argument('output-dir')
+@click.argument('concepts-filename')
+@click.option('-l', '--language', default='en')
+@click.option('--tree-depth', default=1000)
+@click.option('-v', '--verbose', is_flag=True)
+def make_save_replacements(frame_filename, output_dir, concepts_filename, language, tree_depth,
+                           verbose):
+    frame = pd.read_hdf(frame_filename, 'mat', encoding='utf-8')
+    big_frame = make_big_frame(frame, language)
+    small_frame = make_small_frame(big_frame, concepts_filename)
+    replacements = make_replacements_faster(small_frame, big_frame, tree_depth, verbose)
+    save_replacements(path.join(output_dir, '{}_replacements.msgpack'.format(language)),
+                      replacements)
+    save_frame(path.join(output_dir, '{}_big_frame.h5'.format(language)), big_frame)
+    save_frame(path.join(output_dir, '{}_small_frame.h5'.format(language)), small_frame)
+
+if __name__ == '__main__':
```

Comment:
It doesn't make sense to me that running `transforms.py` as a script means to run `make_save_replacements`. This is the kind of thing that should be one of the commands in `cli.py`.